### PR TITLE
Custom permissions + authentication dependent on MDM managed device/app settings

### PR DESCRIPTION
**Pull request by wmathurin**
Reading custom permissions from identity response / storing them alongside the custom settings
Erring out authentication if manage app perm is on but app is not managed

No tests yet
Custom settings and custom permissions are stored together in the same pref file, so if the same keys are used, they will conflict

### CODE REVIEW

File: libs/SalesforceSDK/src/com/salesforce/androidsdk/rest/AdminPrefsManager.java [hunk 1]
```diff
@@ -83,27 +78,18 @@ public void setPrefs(JSONObject attribs, UserAccount account) {
...
-	        e.commit();
-	    }
+        setPrefs(new JSONObject(attribs), account);
 	}
 
-	/**
+    private SharedPreferences getSharedPreferences(UserAccount account) {
+        String sharedPrefPath = ADMIN_PREFS;
+        if (account != null) {
+            sharedPrefPath = ADMIN_PREFS + account.getOrgLevelFilenameSuffix();
```
Comment (bhariharan) [inline libs/SalesforceSDK/src/com/salesforce/androidsdk/rest/AdminPrefsManager.java]: Are the ```custom_permissions``` user level or org level settings?

**Comment by bhariharan** [comment]
Looks good otherwise.

File: libs/SalesforceSDK/src/com/salesforce/androidsdk/rest/AdminPrefsManager.java (hunk 1 above)
Comment (wmathurin) [inline libs/SalesforceSDK/src/com/salesforce/androidsdk/rest/AdminPrefsManager.java]: There are user level.

Comment (bhariharan) [inline libs/SalesforceSDK/src/com/salesforce/androidsdk/rest/AdminPrefsManager.java]: Then we need to use ```account.getUserLevelFilenameSuffix()``` here. And if we're doing this across the board, we'll need a one time upgrade step to migrate an existing org level admin pref file to user level :-(

Comment (wmathurin) [inline libs/SalesforceSDK/src/com/salesforce/androidsdk/rest/AdminPrefsManager.java]: Either that, or maybe I should use other pref files for the custom perms (and they would use the user level filename prefix).

Comment (wmathurin) [inline libs/SalesforceSDK/src/com/salesforce/androidsdk/rest/AdminPrefsManager.java]: Maybe I should create a AdminPermsManager to store custom permissions?

Comment (bhariharan) [inline libs/SalesforceSDK/src/com/salesforce/androidsdk/rest/AdminPrefsManager.java]: Yeah, that's probably easier to maintain. Works for me.

**Comment by wmathurin** [comment]
Refactored AdminPrefsManager / added AdminPermsManager - see https://github.com/wmathurin/SalesforceMobileSDK-Android/commit/cb6b945b9ef310c0d89a0885efcbcc720cc7f690

**Comment by bhariharan** [comment]
Looks good to me. :+1:

**Pull request merged**